src/mcts/search.cc

```diff
@@ -956,7 +956,12 @@ SearchWorker::NodeToProcess SearchWorker::PickNodeToExtend(
         ++possible_moves;
       }
       const float Q = child.GetQ(fpu);
-      const float score = child.GetU(puct_mult) + Q;
+      float score = child.GetU(puct_mult);
+      if (Q == 1 || Q == -1) {
+        score = Q;
+      } else if (Q != 0) {
+        score = tanh ( score + 0.5 * log ( (1 + Q) / (1 - Q) ) );
```

Comment:
just wanted to point out that function for computing score is equivalent to
https://www.wolframalpha.com/input/?i=(-1+-+(e%5E(2+y)+(1+%2B+x))%2F(-1+%2B+x))%2F(1+-+(e%5E(2+y)+(1+%2B+x))%2F(-1+%2B+x))+for+x+in+-1+to+1
with y:=score; x:=Q

I'll leave this if there is ever need to optimize this.
I have not compared the numerical stability, though the division is probably the biggest issue here and it has to stay

Reply:
Using `tanh(x+y) = (tanh x + tanh y)/(tanh x tanh y + 1)`, we can simplify to

`score = ( tanh( score ) + Q) / ( tanh( score ) * Q + 1 )`

which only needs one `tanh` evaluation.

Reply:
tanh is painfully slow
http://quick-bench.com/sa29JAi6ODghRHOa8mJcovD3FEY

but actually the more I look at my formula the more I worry about numerical stability
would need to be tested

Reply:
I found a possible faster implementation for tanh here:
<https://www.ipol.im/pub/art/2015/137/article.pdf>

I'm not going to worry about optimizing it right now though